bluesky/callbacks/broker.py

```diff
@@ -33,24 +60,15 @@ def __init__(self, field, *, fs=None, cmap=None, norm=None,
                  limit_func=None, auto_redraw=True, interpolation=None):
         from xray_vision.backend.mpl.cross_section_2d import CrossSection
         import matplotlib.pyplot as plt
-        super().__init__()
-        self.field = field
+        super().__init__(field, fs=fs)
         fig = plt.figure()
         self.cs = CrossSection(fig, cmap, norm,
                  limit_func, auto_redraw, interpolation)
         self.cs._fig.show()
-        if fs is None:
-            import filestore.api as fs
-        self.fs = fs
 
     def event(self, doc):
-        if doc.get('filled', {}).get(self.field) and self.fs is not None:
-            uid = doc['data'][self.field]
-            data = self.fs.retrieve(uid)
-        else:
-            data = doc['data'][self.field]
+        data = super().event(doc)
```

Comment:
Sorry I got so upset by the lackluster conditional in the base class I completely overlooked that.